### ISSUE TITLE
AWS connection not working

### ISSUE THREAD

**Issue by JGVD21** [jziolkowski/tdm#166]
Using this method [https://tasmota.github.io/docs/AWS-IoT/](url) the devices work perfectly, but when I try to connect tdm to the broker with the same credentials it times out. 

Tested on MacOS, windows and UBUNTU. 

Any advice would be appreciated, and thank you for the work on an amazing tool.

**Comment by ascillato** [comment]
The AWS credentials you have for Tasmota are just for the Tasmota implementation. Those won't work for anything else. So as to connect to AWS with another software different than Tasmota, you have to create a new thing, then download the certificates, set the correct thing policy to allow it to connect, subscribe and publish, and then use the MQTT TLS connection type. So far MQTT TLS is not supported in TDM. Some people are working on adding that. Check the unmerged PR https://github.com/jziolkowski/tdm/pull/164

**Comment by JGVD21** [comment]
Thank you for the response, I did suspect that to be the case.

For anyone reading this question in the future, this article helps you achieve what is described: [https://aws.amazon.com/blogs/iot/how-to-bridge-mosquitto-mqtt-broker-to-aws-iot/](url)

**Comment by JGVD21** [comment]
Okay, So progress was made, the bridge is running and happy, I can see all the messages when connecting to the local mqtt broker.

So the path sonoff > AWS > Mosquitto > MQTTclient is working
But the path sonoff > AWS > Mosquitto > Tdmgr is **NOT** working

I have elevated the debug log level but honestly it doesn't help, I cannot see if MQQT msges are being received

Does anyone have any advice?

Logs and config attached. (the ip of the host is not localhost, I connected from my windows pc to a linux ine to test all the releases of the exe but it also did not work on linux.


[tdm.log](https://github.com/jziolkowski/tdm/files/7254857/tdm.log)

**Comment by barbudor** [comment]
From the log, TDM received a LWT message from a tasmota device which topic is 84CCA894E416
So it sends a request to the devices cmnd/84CCA894E416 to get it's full topic
Apprently either the command doesn't reach the device (what can you see in the device's log) or the response doesn't reach TDM

Beside I am surprised by the subscribed topics. What have you configured as discovery topics ?
Discovery topics must includes every fulltopics used in your devices. It must be exhaustive and not including any wildcard.

**Comment by JGVD21** [comment]
Thank you again for taking the time to respond. I really appreciate it.

to answer at least one part: I tried EVERYTING when it comes to topics and I don't have any fancy configuration on the full topic. it is stock standard, just the MAC address as topic , the rest is perfect. 

I ended up solving the issue by just editing the code

Warning People might freak out on me for this one, I simply added this line **self.mqtt.subscribe("#")**
And then everything fired up an was happy. I was doing it for debugging, but now it is part of my compiled version until this project includes AWS certificates.

I'd like to thank everyone who contributed to this project. I fired up my first Sonoff yesterday morning, it is now 1am in the morning in my neck of the woods (South Africa), and now I've place an order for over 60 Sonoff devices just because the community support was there to make this process easy.
